sw/device/tests/sim_dv/sleep_pin_mio_dio_val_test.c

```diff
@@ -0,0 +1,165 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rand_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+/**
+ * Test: chip_sw_sleep_pin_mio_dio_val
+ *
+ * chip_sw_sleep_pin_mio_dio_val test is to check the PADs assert desired
+ * values. When the chip enters deep powerdown (turning off power domains
+ * except AON), PADs cannot get the data from off domain logics.
+ *
+ * To not confuse the devices outside of the logic, usually isolation cells are
+ * placed in between (off and on). OpenTitan, on top of the isolation, provides
+ * a functionality to control the PADs output value among **0**, **1**,
+ * **High-Z**, and **Passthrough**. Refer the PINMUX technical specification
+ * for the details.
+ *
+ * In this test, the code randomizes the output values except the
+ * **passthrough**, as **passthrough** is tested in other chip level test. Then
+ * the chosen values are given to the UVM testbenches. UVM testbenches compare
+ * the expected and measured values.
+ */
+
+/* NUM_MIO_PADS */
+/* NUM_DIO_PADS */
+
+/* DioOptOut: Some of DIO PADs are input only or does not need to be tested as
+ * they are alive in deep powerdown state. This lists out the PADs. Its entry
+ * is the index of DIOs.
+ *
+ * ```systemverilog
+ * localparam chip_io_e DioPads [NumDioPads] = '{
+ *   UsbP,       // DIO 0                INOUT
+ *   UsbN,       // DIO 1                INOUT
+ *   SpiHostD0,  // DIO 2                INOUT
+ *   SpiHostD1,  // DIO 3                INOUT
+ *   SpiHostD2,  // DIO 4                INOUT
+ *   SpiHostD3,  // DIO 5                INOUT
+ *   SpiDevD0,   // DIO 6                INOUT
+ *   SpiDevD1,   // DIO 7                INOUT
+ *   SpiDevD2,   // DIO 8                INOUT
+ *   SpiDevD3,   // DIO 9                INOUT
+ *   IoR8,       // DIO 10 EC_RST_L      INOUT
+ *   IoR9,       // DIO 11 FLASH_WP_L    INOUT
+ *   SpiDevClk,  // DIO 12               INPUT
+ *   SpiDevCsL,  // DIO 13               INPUT
+ *   SpiHostClk, // DIO 14               OUTPUT
+ *   SpiHostCsL  // DIO 15               OUTPUT
+ * }
+ * ```
+ */
+#define NUM_OPTOUT_DIO 2
```

Comment:
hmm? which ones are these..?

Reply:
ah okay...these are `spi_dev_clk` and `spi_dev_csb`.
Maybe i can generate a header for this also, so you don't have to hard wire it like this.

Reply:
yep. I didn't use this array to make them explicitly High-Z yet. I will do in following PRs.